template.go

```diff
@@ -193,6 +200,14 @@ const (
 				return mm_atomic.LoadUint64(&m.before{{$method.Name}}Counter)
 			}
 
+			{{ if $method.HasParams }}
+				// Calls returns a list of arguments used in each call to {{$mock}}.{{$method.Name}}.
+				// The list is in the same order as the calls were made (i.e. recent calls have a higher index)
+				func (m *m{{$mock}}{{$method.Name}}) Calls() []*{{$mock}}{{$method.Name}}Params {
+					return m.callArgs
```

Comment:
This line also needs to be protected by mutex. I think sync.RWMutex is fits best here.

Reply:
Fixed. `Calls()` now creates a shallow copy of the list as well